### PR TITLE
Adding logging for deploy

### LINKED ISSUE #24: Log any exceptions during deployment
The calling core Data Flow server code only logs the message from the exception. We should add a catch clause in the Kubernetes deployer and log more detailed exception message including the cause and stacktrace. See https://github.com/spring-cloud/spring-cloud-dataflow-server-kubernetes/issues/81

### PR DESCRIPTION
**Pull request by trisberg**
Fixes #24

### CODE REVIEW

File: src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesAppDeployer.java [hunk 1]
```diff
@@ -76,23 +76,28 @@ public String deploy(AppDeploymentRequest request) {
...
-		logger.debug("Creating repl controller: {} on {}", appId, externalPort);
-		createReplicationController(appId, request, idMap, externalPort);
+			logger.debug("Creating repl controller: {} on {}", appId, externalPort);
+			createReplicationController(appId, request, idMap, externalPort);
 
-		return appId;
+			return appId;
+		} catch (Exception e) {
+			logger.error(e.getMessage(), e);
+			throw e;
```
Comment (markfisher) [inline src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesAppDeployer.java]: shouldn't this be wrapped in a `RuntimeException`?

File: src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesAppDeployer.java [hunk 2]
```diff
@@ -132,9 +137,9 @@ public void undeploy(String appId) {
 			selector.put(SPRING_APP_KEY, appId);
 			Boolean podDeleted = client.pods().withLabels(selector).delete();
 			logger.debug("Deleted pods for: {} {}", appId, podDeleted);
-		} catch (KubernetesClientException e) {
+		} catch (Exception e) {
 			logger.error(e.getMessage(), e);
-			throw new RuntimeException(e);
+			throw e;
```
Comment (markfisher) [inline src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesAppDeployer.java]: shouldn't this be wrapped in a `RuntimeException` here also?

File: src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesAppDeployer.java (hunk 1 above)
Comment (trisberg) [inline src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesAppDeployer.java]: Actually, they would already be RuntimeExceptions, should change the catch to RuntimeException and the just re-throw it. Unless we would want to introduce a new DeploymentException in core.

**Comment by markfisher** [comment]
LGTM; squashed and merged as https://github.com/spring-cloud/spring-cloud-deployer-kubernetes/commit/00848d5ec9ce7cf4f59eced8c0b2c19196876eb4

**Pull request closed without merging**